### PR TITLE
Add Connection_Termination_Details as a CEL property

### PR DESCRIPTION
**Pull request by gargnupur**
Signed-off-by: gargnupur <gargnupur@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: Add Connection_Termination_Details as a CEL property
Additional Description:
Risk Level: low
Testing: unit tests
Docs Changes: yes
Release Notes: yes

### CODE REVIEW

File: source/extensions/filters/common/expr/context.h [hunk 1]
```diff
@@ -54,6 +54,7 @@ constexpr absl::string_view Connection = "connection";
 constexpr absl::string_view MTLS = "mtls";
 constexpr absl::string_view RequestedServerName = "requested_server_name";
 constexpr absl::string_view TLSVersion = "tls_version";
+constexpr absl::string_view ConnectionTerminationDetails = "connection_termination_details";
```
Comment (kyessenov) [inline source/extensions/filters/common/expr/context.h:57]: This should be `termination_details`. Also, please update RBAC docs.

Comment (gargnupur) [reply source/extensions/filters/common/expr/context.h]: Thanks..FIxed

File: docs/root/intro/arch_overview/security/rbac_filter.rst [hunk 2]
```diff
@@ -102,6 +102,7 @@ The following attributes are exposed to the language runtime:
    connection.uri_san_local_certificate, string, The first URI entry in the SAN field of the local certificate in the downstream TLS connection
    connection.uri_san_peer_certificate, string, The first URI entry in the SAN field of the peer certificate in the downstream TLS connection
    connection.id, uint, Downstream connection ID
+   connection.termination_details, string, The termination details of the connection
```
Comment (yangminzhu) [inline docs/root/intro/arch_overview/security/rbac_filter.rst:105]: @kyessenov @gargnupur hmm, I'm not sure if we have separate doc for attributes available in CEL but is it really a good idea to add it to the rbac doc? That makes it part of the rbac API and I'm not sure who would ever want to use `termination_details` in access control?

Comment (kyessenov) [reply docs/root/intro/arch_overview/security/rbac_filter.rst:105]: We'll need to sort it out later. But let's keep here for now in one table. We need to split them into two parts, one used before request processing and one used after. Termination details could be used for auditing, for example.

**Review by kyessenov — DISMISSED** [review]

**Review by lizan — COMMENTED** [review]
release notes?

**Comment by gargnupur** [comment]
> release notes?

Added. PATL.. thanks!

**Review by lizan — DISMISSED** [review]

**Review by lizan — APPROVED** [review]

**Pull request merged**